pkg/compiler/compiler.go

```diff
@@ -262,3 +235,49 @@ func CompileAndSave(src string, o *Options) ([]byte, error) {
 
 	return b, nil
 }
+
+// CreateManifest creates manifest and checks that is is valid.
+func CreateManifest(di *DebugInfo, o *Options) (*manifest.Manifest, error) {
+	m, err := di.ConvertToManifest(o)
+	if err != nil {
+		return m, fmt.Errorf("failed to convert debug info to manifest: %w", err)
+	}
+	if !o.NoStandardCheck {
+		if err := standard.CheckABI(m, o.ContractSupportedStandards...); err != nil {
+			return m, err
+		}
+		if m.ABI.GetMethod(manifest.MethodOnNEP11Payment, -1) != nil {
+			if err := standard.CheckABI(m, manifest.NEP11Payable); err != nil {
+				return m, err
+			}
+		}
+		if m.ABI.GetMethod(manifest.MethodOnNEP17Payment, -1) != nil {
+			if err := standard.CheckABI(m, manifest.NEP17Payable); err != nil {
+				return m, err
+			}
+		}
+	}
+	if !o.NoEventsCheck {
+		for name := range di.EmittedEvents {
```

Comment:
Just a question: then maybe we already can remove events from `yml` configuration and generate them automatically?

Reply:
There is a task for this discussion https://github.com/nspcc-dev/neo-go/issues/1038